test/indentation_test.js

```diff
@@ -52,4 +52,16 @@ describe('indentation', () => {
       });
     `);
   });
+
+  it('handles indented lambda as arg to inline lambda', () => {
+    check(`
+    Q -> P ->
+      foo
+    .then (bar) ->
```

Comment:
It looks like if there's another blank line after the `foo` line, it's still valid CoffeeScript but fails with the previous error. Can you try to make your change general enough to handle that?